File: app/src/main/java/com/example/ui/screens/AccountScreen.kt
package com.example.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.SwapHoriz
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.example.data.model.UserAccount
import com.example.data.model.UserRole
import com.example.ui.theme.LogisticsNavy
import com.example.ui.theme.LogisticsOrange
import com.example.ui.theme.LogisticsTeal
import com.example.ui.viewmodel.ShipTrackerViewModel

@Composable
fun AccountScreen(
    viewModel: ShipTrackerViewModel,
    modifier: Modifier = Modifier
) {
    val currentUser by viewModel.currentUser.collectAsStateWithLifecycle()

    Column(
        modifier = modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(14.dp)
    ) {
        // Module 1 Info Banner
        Card(
            colors = CardDefaults.cardColors(containerColor = LogisticsNavy),
            shape = RoundedCornerShape(14.dp),
            modifier = Modifier.fillMaxWidth()
        ) {
            Column(modifier = Modifier.padding(16.dp)) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Column {
                        Text(
                            text = "Module 1: Authentication & RBAC",
                            color = Color.White,
                            fontSize = 16.sp,
                            fontWeight = FontWeight.Bold
                        )
                        Text(
                            text = "Role-Based Access Control • Profile Management",
                            color = Color(0xFF94A3B8),
                            fontSize = 11.sp
                        )
                    }
                    Box(
                        modifier = Modifier
                            .size(36.dp)
                            .background(LogisticsOrange, CircleShape),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = Icons.Default.Person,
                            contentDescription = null,
                            tint = Color.White,
                            modifier = Modifier.size(20.dp)
                        )
                    }
                }
            }
        }

        // Active Profile Card
        Card(
            colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
            shape = RoundedCornerShape(14.dp),
            elevation = CardDefaults.cardElevation(defaultElevation = 1.dp),
            modifier = Modifier.fillMaxWidth()
        ) {
            Column(modifier = Modifier.padding(16.dp)) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Box(
                        modifier = Modifier
                            .size(54.dp)
                            .background(Color(currentUser.avatarColorHex), CircleShape),
                        contentAlignment = Alignment.Center
                    ) {
                        Text(
                            text = currentUser.fullName.take(1),
                            color = Color.White,
                            fontSize = 22.sp,
                            fontWeight = FontWeight.Bold
                        )
                    }
                    Spacer(modifier = Modifier.width(14.dp))
                    Column {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Text(
                                text = currentUser.fullName,
                                fontSize = 16.sp,
                                fontWeight = FontWeight.Bold
                            )
                            Spacer(modifier = Modifier.width(8.dp))
                            Surface(
                                color = if (currentUser.role == UserRole.OWNER_ADMIN) LogisticsOrange else LogisticsTeal,
                                shape = RoundedCornerShape(4.dp)
                            ) {
                                Text(
                                    text = if (currentUser.role == UserRole.OWNER_ADMIN) "ADMIN" else "STAFF",
                                    color = Color.White,
                                    fontSize = 10.sp,
                                    fontWeight = FontWeight.Bold,
                                    modifier = Modifier.padding(horizontal = 6.dp, vertical = 2.dp)
                                )
                            }
                        }
                        Text(
                            text = currentUser.email,
                            fontSize = 12.sp,
                            color = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                        Text(
                            text = "Staff ID: ${currentUser.id.uppercase()}",
                            fontSize = 11.sp,
                            color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f)
                        )
                    }
                }

                Spacer(modifier = Modifier.height(12.dp))
                Surface(
                    color = MaterialTheme.colorScheme.surfaceVariant,
                    shape = RoundedCornerShape(8.dp),
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Text(
                        text = "Access Scope: ${currentUser.role.description}",
                        fontSize = 11.sp,
                        color = MaterialTheme.colorScheme.onSurface,
                        modifier = Modifier.padding(10.dp)
                    )
                }
            }
        }

        // Switch Active Account / Role (Capstone Evaluation Tool)
        Text(
            text = "Switch Active Account (Capstone Demonstration):",
            fontSize = 13.sp,
            fontWeight = FontWeight.Bold,
            color = MaterialTheme.colorScheme.onSurface
        )

        UserAccount.ALL_ACCOUNTS.forEach { account ->
            val isCurrent = account.id == currentUser.id
            Surface(
                color = if (isCurrent) MaterialTheme.colorScheme.primaryContainer else MaterialTheme.colorScheme.surfaceVariant,
                shape = RoundedCornerShape(10.dp),
                border = androidx.compose.foundation.BorderStroke(
                    1.dp,
                    if (isCurrent) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.outline
                ),
                modifier = Modifier
                    .fillMaxWidth()
                    .clickable { viewModel.switchUser(account) }
                    .testTag("switch_to_${account.id}")
            ) {
                Row(
                    modifier = Modifier.padding(12.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Box(
                        modifier = Modifier
                            .size(34.dp)
                            .background(Color(account.avatarColorHex), CircleShape),
                        contentAlignment = Alignment.Center
                    ) {
                        Text(
                            text = account.fullName.take(1),
                            color = Color.White,
                            fontWeight = FontWeight.Bold
                        )
                    }
                    Spacer(modifier = Modifier.width(10.dp))
                    Column(modifier = Modifier.weight(1f)) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Text(
                                text = account.fullName,
                                fontWeight = FontWeight.Bold,
                                fontSize = 13.sp,
                                color = MaterialTheme.colorScheme.onSurface
                            )
                            Spacer(modifier = Modifier.width(6.dp))
                            Surface(
                                color = if (account.role == UserRole.OWNER_ADMIN) LogisticsOrange else LogisticsTeal,
                                shape = RoundedCornerShape(4.dp)
                            ) {
                                Text(
                                    text = if (account.role == UserRole.OWNER_ADMIN) "ADMIN" else "STAFF",
                                    color = Color.White,
                                    fontSize = 9.sp,
                                    fontWeight = FontWeight.Bold,
                                    modifier = Modifier.padding(horizontal = 4.dp, vertical = 2.dp)
                                )
                            }
                        }
                        Text(
                            text = account.email,
                            fontSize = 11.sp,
                            color = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                    }
                    if (isCurrent) {
                        Icon(imageVector = Icons.Default.Check, contentDescription = "Active", tint = MaterialTheme.colorScheme.primary)
                    }
                }
            }
        }

        // RBAC Permissions Matrix Table
        Card(
            colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
            shape = RoundedCornerShape(12.dp),
            border = androidx.compose.foundation.BorderStroke(1.dp, MaterialTheme.colorScheme.outline),
            modifier = Modifier.fillMaxWidth()
        ) {
            Column(modifier = Modifier.padding(14.dp)) {
                Text(
                    text = "Role-Based Access Control (RBAC) Matrix",
                    fontWeight = FontWeight.Bold,
                    fontSize = 13.sp,
                    color = MaterialTheme.colorScheme.onSurface
                )
                Spacer(modifier = Modifier.height(10.dp))

                // Header
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .background(MaterialTheme.colorScheme.surfaceVariant, RoundedCornerShape(6.dp))
                        .padding(horizontal = 8.dp, vertical = 6.dp),
                    horizontalArrangement = Arrangement.SpaceBetween
                ) {
                    Text(
                        text = "Operational Capability",
                        fontSize = 11.sp,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.onSurface,
                        modifier = Modifier.weight(2f)
                    )
                    Text(
                        text = "Admin",
                        fontSize = 11.sp,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.onSurface,
                        modifier = Modifier.weight(1f)
                    )
                    Text(
                        text = "Staff",
                        fontSize = 11.sp,
                        fontWeight = FontWeight.Bold,
                        color = MaterialTheme.colorScheme.onSurface,
                        modifier = Modifier.weight(1f)
                    )
                }

                val matrix = listOf(
                    Triple("Scan & Register Parcels", true, true),
                    Triple("Auto-Classify Courier & Platform", true, true),
                    Triple("Update Lifecycle Status (SOP)", true, true),
                    Triple("Create Dispatch Batch Handover", true, true),
                    Triple("Log Returns & Damage Appraisal", true, true),
                    Triple("Offline Cache & Sync", true, true),
                    Triple("Export CSV & Dispatch Reports", true, true),
                    Triple("View Financial / Declared Value", true, false),
                    Triple("Delete Shipments from Registry", true, false),
                    Triple("Modify Courier Agreements", true, false)
                )

                matrix.forEach { (feature, adminAllowed, staffAllowed) ->
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(horizontal = 8.dp, vertical = 6.dp),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Text(
                            text = feature,
                            fontSize = 11.sp,
                            color = MaterialTheme.colorScheme.onSurface,
                            modifier = Modifier.weight(2f)
                        )
                        Box(modifier = Modifier.weight(1f)) {
                            if (adminAllowed) {
                                Icon(imageVector = Icons.Default.Check, contentDescription = "Yes", tint = Color(0xFF059669), modifier = Modifier.size(16.dp))
                            } else {
                                Icon(imageVector = Icons.Default.Close, contentDescription = "No", tint = Color(0xFFDC2626), modifier = Modifier.size(16.dp))
                            }
                        }
                        Box(modifier = Modifier.weight(1f)) {
                            if (staffAllowed) {
                                Icon(imageVector = Icons.Default.Check, contentDescription = "Yes", tint = Color(0xFF059669), modifier = Modifier.size(16.dp))
                            } else {
                                Icon(imageVector = Icons.Default.Close, contentDescription = "No", tint = Color(0xFFDC2626), modifier = Modifier.size(16.dp))
                            }
                        }
                    }
                }
            }
        }

        Spacer(modifier = Modifier.height(20.dp))
    }
}

File: flutter_app/lib/screens/account_screen.dart
import 'package:flutter/material.dart';
import '../services/logistics_service.dart';
import '../models/user_account.dart';
import '../models/enums.dart';

class AccountScreen extends StatelessWidget {
  final LogisticsService service;

  const AccountScreen({super.key, required this.service});

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    final currentUser = service.currentUser;

    return Scaffold(
      appBar: AppBar(
        title: const Text('Staff & RBAC Security Profile', style: TextStyle(fontWeight: FontWeight.bold, fontSize: 16)),
      ),
      body: ListView(
        padding: const EdgeInsets.all(16),
        children: [
          // Current User Profile Card
          Card(
            elevation: 0,
            shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(12),
              side: BorderSide(color: theme.colorScheme.outline.withOpacity(0.3)),
            ),
            child: Padding(
              padding: const EdgeInsets.all(16),
              child: Column(
                children: [
                  CircleAvatar(
                    radius: 30,
                    backgroundColor: currentUser.avatarColor,
                    child: Text(
                      currentUser.fullName.substring(0, 1),
                      style: const TextStyle(color: Colors.white, fontSize: 24, fontWeight: FontWeight.bold),
                    ),
                  ),
                  const SizedBox(height: 10),
                  Text(currentUser.fullName, style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 16)),
                  const SizedBox(height: 2),
                  Container(
                    padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 2),
                    decoration: BoxDecoration(
                      color: currentUser.role == UserRole.ownerAdmin ? Colors.blue.shade50 : Colors.teal.shade50,
                      borderRadius: BorderRadius.circular(12),
                    ),
                    child: Text(
                      currentUser.role.displayName,
                      style: TextStyle(
                        color: currentUser.role == UserRole.ownerAdmin ? Colors.blue.shade900 : Colors.teal.shade900,
                        fontWeight: FontWeight.bold,
                        fontSize: 11,
                      ),
                    ),
                  ),
                  const SizedBox(height: 6),
                  Text(currentUser.email, style: TextStyle(fontSize: 12, color: theme.colorScheme.onSurfaceVariant)),
                  Text('Staff Code: ${currentUser.staffCode}', style: TextStyle(fontSize: 11, color: theme.colorScheme.onSurfaceVariant)),
                  const SizedBox(height: 12),
                  Container(
                    padding: const EdgeInsets.all(10),
                    decoration: BoxDecoration(
                      color: theme.colorScheme.surfaceVariant,
                      borderRadius: BorderRadius.circular(8),
                    ),
                    child: Text(
                      'Access Scope: ${currentUser.role.description}',
                      style: const TextStyle(fontSize: 11),
                      textAlign: TextAlign.center,
                    ),
                  ),
                ],
              ),
            ),
          ),
          const SizedBox(height: 16),

          // Switch Account Demonstration
          Text('Switch Active Account (RBAC Demo)', style: theme.textTheme.titleMedium?.copyWith(fontWeight: FontWeight.bold)),
          const SizedBox(height: 8),
          ...UserAccount.allAccounts.map((account) {
            final isCurrent = account.id == currentUser.id;
            return Card(
              elevation: 0,
              margin: const EdgeInsets.only(bottom: 8),
              color: isCurrent ? theme.colorScheme.primaryContainer : theme.colorScheme.surface,
              shape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(10),
                side: BorderSide(color: isCurrent ? theme.colorScheme.primary : theme.colorScheme.outline.withOpacity(0.3)),
              ),
              child: ListTile(
                onTap: () {
                  service.switchAccount(account);
                  ScaffoldMessenger.of(context).showSnackBar(
                    SnackBar(content: Text('Switched to ${account.fullName}')),
                  );
                },
                leading: CircleAvatar(
                  backgroundColor: account.avatarColor,
                  child: Text(account.fullName.substring(0, 1), style: const TextStyle(color: Colors.white, fontWeight: FontWeight.bold)),
                ),
                title: Text(account.fullName, style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 13)),
                subtitle: Text('${account.role.displayName} • ${account.email}', style: const TextStyle(fontSize: 11)),
                trailing: isCurrent ? const Icon(Icons.check_circle, color: Colors.green) : const Icon(Icons.chevron_right),
              ),
            );
          }),
          const SizedBox(height: 16),

          // RBAC Matrix Table
          Card(
            elevation: 0,
            shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(12),
              side: BorderSide(color: theme.colorScheme.outline.withOpacity(0.3)),
            ),
            child: Padding(
              padding: const EdgeInsets.all(14),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  const Text('Role-Based Access Control (RBAC) Matrix', style: TextStyle(fontWeight: FontWeight.bold, fontSize: 13)),
                  const SizedBox(height: 10),
                  _buildMatrixHeader(),
                  const Divider(height: 12),
                  _buildMatrixRow('Barcode Scanner Intake', true, true),
                  _buildMatrixRow('Status Update & Lifecycle', true, true),
                  _buildMatrixRow('Dispatch Batch Manifest', true, true),
                  _buildMatrixRow('Log Return-To-Sender (RTS)', true, true),
                  _buildMatrixRow('View Gross Financials', true, false),
                  _buildMatrixRow('Permanent Record Deletion', true, false),
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildMatrixHeader() {
    return const Row(
      children: [
        Expanded(flex: 3, child: Text('Operational Capability', style: TextStyle(fontWeight: FontWeight.bold, fontSize: 11))),
        Expanded(child: Center(child: Text('Admin', style: TextStyle(fontWeight: FontWeight.bold, fontSize: 11)))),
        Expanded(child: Center(child: Text('Staff', style: TextStyle(fontWeight: FontWeight.bold, fontSize: 11)))),
      ],
    );
  }

  Widget _buildMatrixRow(String feature, bool adminAllowed, bool staffAllowed) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 4),
      child: Row(
        children: [
          Expanded(flex: 3, child: Text(feature, style: const TextStyle(fontSize: 11))),
          Expanded(
            child: Center(
              child: Icon(
                adminAllowed ? Icons.check : Icons.close,
                size: 16,
                color: adminAllowed ? Colors.green : Colors.red,
              ),
            ),
          ),
          Expanded(
            child: Center(
              child: Icon(
                staffAllowed ? Icons.check : Icons.close,
                size: 16,
                color: staffAllowed ? Colors.green : Colors.red,
              ),
            ),
          ),
        ],
      ),
    );
  }
}
